src/application-manager.coffee

```diff
@@ -0,0 +1,875 @@
+				return serviceAction('kill', target.serviceId, current, target)
+			else
+				return null
+		'kill-then-download': (current, target) ->
+			return serviceAction('kill', target.serviceId, current, target)
+		'delete-then-download': (current, target, needsDownload) ->
+			return serviceAction('kill', target.serviceId, current, target, removeImage: needsDownload)
+		'hand-over': (current, target, needsDownload, dependenciesMetForStart, dependenciesMetForKill, needsSpecialKill, timeout) ->
+			if needsDownload
+				return fetchAction(target)
+			else if needsSpecialKill && dependenciesMetForKill()
+				return serviceAction('kill', target.serviceId, current, target)
+			else if dependenciesMetForStart()
+				return serviceAction('handover', target.serviceId, current, target, timeout: timeout)
+			else
+				return null
+	}
+
+	_nextStepForService: ({ current, target }, updateContext) ->
+		{ targetApp, networkPairs, volumePairs, installPairs, updatePairs, stepsInProgress, availableImages } = updateContext
+		if _.find(stepsInProgress, (step) -> step.serviceId == target.serviceId)?
+			# There is already a step in progress for this service, so we wait
+			return null
+
+		needsDownload = !_.some(availableImages, (image) -> target.image == image.name)
+		dependenciesMetForStart = =>
+			@_dependenciesMetForServiceStart(target, networkPairs, volumePairs, installPairs.concat(updatePairs), stepsInProgress)
+		dependenciesMetForKill = =>
+			!needsDownload and @_dependenciesMetForServiceKill(target, targetApp, availableImages)
+
+		# If the service is using a network or volume that is being updated, we need to kill it
+		# even if its strategy is handover
+		needsSpecialKill = @_hasCurrentNetworksOrVolumes(current, networkPairs, volumePairs)
+
+		if current?.isSameContainer(target)
+			# We're only stopping/starting it
+			return @_updateContainerStep(current, target)
+		else if !current?
+			# Either this is a new service, or the current one has already been killed
+			return @_fetchOrStartStep(current, target, needsDownload, dependenciesMetForStart)
+		else
+			strategy = checkString(target.labels['io.resin.update.strategy'])
+			validStrategies = [ 'download-then-kill', 'kill-then-download', 'delete-then-download', 'hand-over' ]
+			strategy = 'download-then-kill' if !_.includes(validStrategies, strategy)
+			timeout = checkInt(target.labels['io.resin.update.handover_timeout'])
+			return @_strategySteps[strategy](current, target, needsDownload, dependenciesMetForStart, dependenciesMetForKill, needsSpecialKill, timeout)
+
+	_nextStepsForAppUpdate: (currentApp, targetApp, availableImages = [], stepsInProgress = []) =>
+		emptyApp = { services: [], volumes: {}, networks: {} }
+		if !targetApp?
+			targetApp = emptyApp
+		if !currentApp?
+			currentApp = emptyApp
+		appId = targetApp.appId ? currentApp.appId
+		# Create the default network for the target app
+		targetApp.networks[targetApp.appId] ?= {}
```

Comment:
If the targetApp is undefined then this will create a network?

Reply:
Good catch, and Shaun was actually hitting this. Fixed.